File: Matlab/PDE/FDM/SBPSAT/HyperbolicSystem2D/HyperbolicSystem2DSolver.m
function [s, t] = HyperbolicSystem2DSolver(problem, acc, data, N, minv, maxv, geometry)

%% Initializing grid
grid          = grid_obj;
grid.minv     = minv;
grid.maxv     = maxv;
grid.N        = N;
grid.geometry = geometry;
grid          = createGrid(grid);
grid.plotGrid()

%% Initializing problem
p = problem_obj;
p.type.problem = problem;
p.type.data    = data;
p = initialize(p);
p = createMatrices(p, grid);
p = createData(p);

%% Initializing SBP operators
sbp     = sbp_obj;
sbp.acc = acc;
sbp     = createOperators(sbp, grid);

%% Initializing scheme
s = scheme_obj;
s = createBoundaryOperator(s, grid, sbp, p);
s = createPenalties(s);
s = createAMatrix(s, grid, sbp, p);
s = createScheme(s, grid, p);
tic
s = solveUsingRK4(s, grid, p);
t = toc;
s = computeError(s, grid, sbp, p);

File: Python/Differential Equations/PDE/SBPSAT/HyperbolicSystem2D/HyperbolicSystem2DSolver.py
from GridClass import GridClass
from ProblemClass import ProblemClass
from SBPClass import SBPClass
from SchemeClass import SchemeClass

g = GridClass([0, 0, 0], [1, 1, 1], [10, 10, 1000])
#g.plot_grid()

p = ProblemClass('Maxwell', g)
p.createData(g)

sbp     = SBPClass(g)
sbp.acc = 2

s = SchemeClass(p)
s.ContinuousBoundaryOperators()
s.numericalBoundaryOperators(g, p)
s.createPenalties()
s.CreateMatrices(g, sbp, p)
s.Scheme(g, sbp, p)
s.Error(sbp, g, p, 1)
s.plotSolution(g, p)
